arrow/arrow.py

```diff
@@ -1445,84 +1558,140 @@ def for_json(self):
 
     # math
 
-    def __add__(self, other):
+    @overload
+    def __add__(self, other: timedelta) -> "Arrow":
+        pass  # pragma: no cover
+
+    @overload
+    def __add__(self, other: relativedelta) -> "Arrow":
+        pass  # pragma: no cover
+
+    def __add__(self, other: Any) -> "Arrow":
 
         if isinstance(other, (timedelta, relativedelta)):
             return self.fromdatetime(self._datetime + other, self._datetime.tzinfo)
 
         return NotImplemented
 
-    def __radd__(self, other):
+    def __radd__(self, other: Union[timedelta, relativedelta]) -> "Arrow":
         return self.__add__(other)
 
-    def __sub__(self, other):
+    @overload
+    def __sub__(self, other: Union[timedelta, relativedelta]) -> "Arrow":
+        pass  # pragma: no cover
+
+    @overload
+    def __sub__(self, other: Union[dt_datetime, "Arrow"]) -> timedelta:
+        pass  # pragma: no cover
+
+    def __sub__(self, other: Any) -> Union[timedelta, "Arrow"]:
 
         if isinstance(other, (timedelta, relativedelta)):
             return self.fromdatetime(self._datetime - other, self._datetime.tzinfo)
 
-        elif isinstance(other, datetime):
+        elif isinstance(other, dt_datetime):
             return self._datetime - other
 
         elif isinstance(other, Arrow):
             return self._datetime - other._datetime
 
         return NotImplemented
 
-    def __rsub__(self, other):
+    @overload
+    def __rsub__(self, other: dt_datetime) -> timedelta:
+        pass  # pragma: no cover
 
-        if isinstance(other, datetime):
+    @overload
+    def __rsub__(self, other: Any) -> timedelta:
+        pass  # pragma: no cover
```

Comment:
What's the purpose of this overload? Isn't it identical to the original `__rsub__` definition below?

Reply:
Originally I want to express `NotImplemented` path in type annotation. but currently python typing does not support this feature. (https://github.com/python/mypy/issues/4791) And I found [related work](https://github.com/pantsbuild/pants/pull/11506#discussion_r567252681) so I remove all `overload`s except `__sub__`.